## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 512134`
Commit: `eee7ac042d2ef25bb9883871299e2113113365f7`

---
### `x86_64-linux`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>envoy</li>
    <li>pomerium</li>
    <li>xla</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 35 packages built:</summary>
  <ul>
    <li>anytype</li>
    <li>anytype-heart</li>
    <li>bant</li>
    <li>fcitx5-mozc</li>
    <li>fcitx5-mozc-ut</li>
    <li>ibus-engines.mozc</li>
    <li>ibus-engines.mozc-ut</li>
    <li>mozc</li>
    <li>mozc-ut</li>
    <li>perf_data_converter</li>
    <li>protoc-gen-js</li>
    <li>python313Packages.arviz</li>
    <li>python313Packages.arviz.dist</li>
    <li>python313Packages.bambi</li>
    <li>python313Packages.bambi.dist</li>
    <li>python313Packages.corner</li>
    <li>python313Packages.corner.dist</li>
    <li>python313Packages.distrax</li>
    <li>python313Packages.distrax.dist</li>
    <li>python313Packages.flowjax</li>
    <li>python313Packages.flowjax.dist</li>
    <li>python313Packages.numpyro</li>
    <li>python313Packages.numpyro.dist</li>
    <li>python313Packages.nutpie</li>
    <li>python313Packages.nutpie.dist</li>
    <li>python313Packages.pymc</li>
    <li>python313Packages.pymc.dist</li>
    <li>python313Packages.pytensor</li>
    <li>python313Packages.pytensor.dist</li>
    <li>python313Packages.rlax</li>
    <li>python313Packages.rlax.dist</li>
    <li>python313Packages.tensorflow-probability</li>
    <li>python313Packages.tensorflow-probability.dist</li>
    <li>verible</li>
    <li>veridian</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>envoy</li>
    <li>pomerium</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 35 packages built:</summary>
  <ul>
    <li>anytype</li>
    <li>anytype-heart</li>
    <li>bant</li>
    <li>fcitx5-mozc</li>
    <li>fcitx5-mozc-ut</li>
    <li>ibus-engines.mozc</li>
    <li>ibus-engines.mozc-ut</li>
    <li>mozc</li>
    <li>mozc-ut</li>
    <li>perf_data_converter</li>
    <li>protoc-gen-js</li>
    <li>python313Packages.arviz</li>
    <li>python313Packages.arviz.dist</li>
    <li>python313Packages.bambi</li>
    <li>python313Packages.bambi.dist</li>
    <li>python313Packages.corner</li>
    <li>python313Packages.corner.dist</li>
    <li>python313Packages.distrax</li>
    <li>python313Packages.distrax.dist</li>
    <li>python313Packages.flowjax</li>
    <li>python313Packages.flowjax.dist</li>
    <li>python313Packages.numpyro</li>
    <li>python313Packages.numpyro.dist</li>
    <li>python313Packages.nutpie</li>
    <li>python313Packages.nutpie.dist</li>
    <li>python313Packages.pymc</li>
    <li>python313Packages.pymc.dist</li>
    <li>python313Packages.pytensor</li>
    <li>python313Packages.pytensor.dist</li>
    <li>python313Packages.rlax</li>
    <li>python313Packages.rlax.dist</li>
    <li>python313Packages.tensorflow-probability</li>
    <li>python313Packages.tensorflow-probability.dist</li>
    <li>verible</li>
    <li>veridian</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>protoc-gen-js</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>protoc-gen-js</li>
    <li>verible</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>protoc-gen-js</summary>
<pre>ERROR: /nix/var/nix/builds/nix-6375-334097090/output/external/com_google_protobuf/third_party/utf8_range/BUILD.bazel:32:11: Compiling third_party/utf8_range/utf8_range.c failed: (Exit 1): cc_wrapper.sh failed: error executing CppCompile command (from target @@com_google_protobuf//third_party/utf8_range:utf8_range) external/local_config_cc/cc_wrapper.sh -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics ... (remaining 61 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from external/com_google_protobuf/third_party/utf8_range/utf8_range.c:18:
external/com_google_protobuf/third_party/utf8_range/utf8_range.h:4:10: error: module com_google_protobuf//third_party/utf8_range:utf8_range does not depend on a module exporting &#x27;stddef.h&#x27;
    4 | #include &lt;stddef.h&gt;
      |          ^
external/com_google_protobuf/third_party/utf8_range/utf8_range.c:20:10: error: module com_google_protobuf//third_party/utf8_range:utf8_range does not depend on a module exporting &#x27;stddef.h&#x27;
   20 | #include &lt;stddef.h&gt;
      |          ^
external/com_google_protobuf/third_party/utf8_range/utf8_range.c:21:10: error: module com_google_protobuf//third_party/utf8_range:utf8_range does not depend on a module exporting &#x27;stdint.h&#x27;
   21 | #include &lt;stdint.h&gt;
      |          ^
3 errors generated.
Target //generator:protoc-gen-js failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 24.044s, Critical Path: 2.20s
INFO: 1113 processes: 1093 internal, 20 processwrapper-sandbox.
ERROR: Build did NOT complete successfully
</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>protoc-gen-js</summary>
<pre>
Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from external/com_google_protobuf/third_party/utf8_range/utf8_range.c:18:
external/com_google_protobuf/third_party/utf8_range/utf8_range.h:4:10: error: module com_google_protobuf//third_party/utf8_range:utf8_range does not depend on a module exporting &#x27;stddef.h&#x27;
    4 | #include &lt;stddef.h&gt;
      |          ^
external/com_google_protobuf/third_party/utf8_range/utf8_range.c:20:10: error: module com_google_protobuf//third_party/utf8_range:utf8_range does not depend on a module exporting &#x27;stddef.h&#x27;
   20 | #include &lt;stddef.h&gt;
      |          ^
external/com_google_protobuf/third_party/utf8_range/utf8_range.c:21:10: error: module com_google_protobuf//third_party/utf8_range:utf8_range does not depend on a module exporting &#x27;stdint.h&#x27;
   21 | #include &lt;stdint.h&gt;
      |          ^
3 errors generated.
[636 / 1,452] [Prepa] Linking external/com_google_protobuf/third_party/utf8_range/libutf8_range.a [for tool] ... (2 actions, 1 running)
Target //generator:protoc-gen-js failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 20.794s, Critical Path: 0.82s
INFO: 638 processes: 637 internal, 1 processwrapper-sandbox.
ERROR: Build did NOT complete successfully
</pre>
</details>
</details>

--- a/pkgs/build-support/build-bazel-package/default.nix
+++ b/pkgs/build-support/build-bazel-package/default.nix
@@ -86,6 +86,8 @@ let
       targetRunFlags ? [ ],
     }:
     lib.optionalString (targets != [ ]) ''
+      concatTo bazelFlagsArray bazelFlags
+
       # See footnote called [USER and BAZEL_USE_CPP_ONLY_TOOLCHAIN variables]
       BAZEL_USE_CPP_ONLY_TOOLCHAIN=1 \
       USER=homeless-shelter \
@@ -99,7 +101,7 @@ let
         "''${host_copts[@]}" \
         "''${linkopts[@]}" \
         "''${host_linkopts[@]}" \
-        $bazelFlags \
+        "''${bazelFlagsArray[@]}" \
         ${lib.strings.concatStringsSep " " additionalFlags} \
         ${lib.strings.concatStringsSep " " targets} \
         ${
@@ -239,13 +241,24 @@ stdenv.mkDerivation (
           );
 
         dontFixup = true;
-        allowedRequisites = [ ];
 
         inherit (lib.fetchers.normalizeHash { hashTypes = [ "sha256" ]; } fetchAttrs)
           outputHash
           outputHashAlgo
           ;
       }
+      // (
+        if fFetchAttrs.__structuredAttrs or false then
+          {
+            # With __structuredAttrs = true, the build always fails with “output $out is not allowed to refer to the following paths: $out”.
+            # This appears to be the same issue as in 283bca9648fc1afb01d3e4c3b5919251429da907.
+            outputChecks.out.allowedRequisites = [ "out" ];
+          }
+        else
+          {
+            allowedRequisites = [ ];
+          }
+      )
     );
 
     nativeBuildInputs = fBuildAttrs.nativeBuildInputs or [ ] ++ [

--- a/pkgs/by-name/ba/bant/package.nix
+++ b/pkgs/by-name/ba/bant/package.nix
@@ -73,6 +73,9 @@ buildBazelPackage rec {
 
   passthru.updateScript = nix-update-script { };
 
+  strictDeps = true;
+  __structuredAttrs = true;
+
   meta = {
     description = "Bazel/Build Analysis and Navigation Tool";
     homepage = "http://bant.build/";

